### ISSUE TITLE
Config file por HP Pavilion Gaming Laptop 15 dk-1000

### ISSUE THREAD

**Issue by StarMember74** [UraniumDonut/nbfc-revive#93]
Hello, I need help with a config file for my pavilion, the exact model is dk-1027la
I've tried all the config files that currently are but none fits to my laptop. I only get the cpu fan to work and it always push ups to 100%. Even If I slide to 0%, it keeps spinning loudly. Maybe someone can help me with my model, I would appreciate it.

**Comment by dzid26** [comment]
Use `ec-probe dump` from `C:\Program Files (x86)\NoteBook FanControl`.
See which registers are 0xFF. Try to change them to 0 and see what happens.  
On HP spectre FF was indicating Automatic fan control. https://github.com/UraniumDonut/nbfc-revive/issues/87

**Comment by StarMember74** [comment]
Hmmm I can't seem to find that ec-probe dump. All I can find in my folder is ec-probe.exe but it does nothing

**Comment by dzid26** [comment]
I mean, you should run it from the terminal

**Comment by StarMember74** [comment]
I'm new into this, so I don't know nothing. Help me, please. What should I do? I opened the exe from cmd...


NoteBook FanControl>ec-probe.exe
NoteBook FanControl EC probing tool

usage: ec-probe.exe [--version] [--help] <command> [<args>]

commands:
  dump                         Dump all EC registers

  read <register>              Read a byte from a EC register

  write <register> <value> [options]
                               Write a byte to a EC register
    -v, --verbose                Be verbose

  monitor [options]            Monitor all EC registers for changes
    -t, --timespan <seconds>     Monitored timespan (default: infinite)
    -i, --interval <seconds>     Set poll interval (default: 5)
    -r, --report <path>          Save all readings as CSV file
    -c, --clearly                Blanks out consecutive duplicate readings
    -d, --decimal                Output readings in decimal format instead of hexadecimal format

All input values are interpreted as decimal numbers by default.
Hexadecimal values may be entered by prefixing them with "0x".

**Comment by StarMember74** [comment]
![Screenshot](https://user-images.githubusercontent.com/109440134/179427986-903e5222-e30c-48dd-b2b9-b3b07d7b7824.png)
How and wich ones do I edit?

**Comment by dzid26** [comment]
There is some help in: `ec-probe.exe`

Try `ec-probe.exe write 0x52 0x0` for that one to change:
<img width="401" alt="image" src="https://user-images.githubusercontent.com/841061/179428800-1aaa573e-58c5-4fc3-8cb3-7dd74dd9fbc8.png">
If nothing happens, try another 0xFF to change to 0x0.  After each time `dump` the values to see if the `write` was even accepted.

**Comment by YousefAhmed2812** [comment]
@StarMember74 were you able to make a working config?

**Comment by StarMember74** [comment]
No man, sorry. I've been really busy. Besides, Seems a little bit complicated for me. If by any chance you can help me, I would appreciate it.

**Comment by YousefAhmed2812** [comment]
to be honest, this is also very complicated for me. I tried doing what @dzid26 said, I changed the FF values to 0x0 but it either does nothing or it doesn't accept it and i couldn't do anything else but there is a working config I don't know if you saw it below that you can use to control the fan you can choose the speed, not just 0 or 100 but it makes the GPU fan go to 100%
https://www.mediafire.com/file/22s30kjgt5mz5m7/HP_Pavilion_Gaming_Laptop_15-dk1xxx.xml/file
you also need to download this dll file https://github.com/UraniumDonut/nbfc-revive/issues/27#issuecomment-967684864
and put it in the nbfc folder in the plugins folder for the program to work properly.

**Comment by StarMember74** [comment]
Oh I see, I tried that profile you mentioned and indeed, it works. I'm assuming the left fan is the gpu one, right? And the right one is for cpu? Let's hope someone that knows about this can lend us a hand with this.

**Comment by YousefAhmed2812** [comment]
yes, the left fan is the GPU fan and the right fan is the CPU fan. apparently, the GPU reading is 15 but I think that's as far as we know.

**Comment by StarMember74** [comment]
Alright, now I know in case sometime I decide to edit something. Lets just wait around, maybe someone with knowledge will help. Take care

**Comment by dzid26** [comment]
> yes, the left fan is the GPU fan and the right fan is the CPU fan. apparently, the GPU reading is 15 but I think that's as far as we know.
 
Seems like you are almost there.  

The XML points to two write registers.  
Fan command speed register is <WriteRegister>20</WriteRegister>
So in hex that 0x14.  (That is the first 0xFF's in the screenshoted table). 

Other register <Register>15</Register> (0x0F) is probably mode, because they are writing to it either 8 or 5. Maybe that's is causing GPU fan go mad. 

Try different values there.  
Or just modify 0x14 register without touching 0x0F one. 

Good luck

**Comment by StarMember74** [comment]
Could you try this?, @YousefAhmed2812 You seem to know a bit more than me and you've already tried modifying some values. Check the upper comment.

**Comment by YousefAhmed2812** [comment]
I am not really sure what @dzid26 said i didn't really understand so I'm not sure what I am supposed to do

**Comment by StarMember74** [comment]
wanna try it, @YousefAhmed2812 ? Seems like we are getting it this time.

**Comment by NeuroCPP** [comment]
try `HP Compaq 15-s103tx` config it seems to work mostly for hp pavilion 15 gaming dk-******
also try to fiddle with these settings
![Screenshot 2023-06-02 204815](https://github.com/UraniumDonut/nbfc-revive/assets/85978957/6a89cf8e-acce-44e7-bf40-bb7ad190e69d)

**Comment by MrPino420** [comment]
I did all you did in here, and i got the cpu fan to work, but the GPU fan goes haywire as soon as the cpu fan speed goes above 10%, any ideas?

**Comment by MrPino420** [comment]
When i change the register 15 (0x0F), the gpu fan changes

> > I am not really sure what @dzid26 said I didn't really understand so I'm not sure what I am supposed to do
> 
> I was referring to the file you linked. I was describing that it points to two write registers. You can either try to manually individually change the registers using ec-probe.exe and see what happens or modify the config XML file using nfbc config editor.
> 
> That register 15 (or 0x0F in hex) looks like it could be the culprit. You can try to turn delete changing it on initialization or try different values. ![image](https://user-images.githubusercontent.com/841061/183404535-f42691b7-63bf-4e36-95e2-fcaaebcab9b2.png)

**Comment by MrPino420** [comment]
If you create a fan configuration for the register 15 (0x0F) you can somewhat control the gpu fan, but there is some issues with the cpu fan tho, it seems they dont work well with each other, this is a matter of tinkering and brute forcing from now on
![image](https://github.com/UraniumDonut/nbfc-revive/assets/86311213/a8977edc-88d7-4c06-9001-0ed80fb5c231)

**Comment by crash725** [comment]
hey guys ive been looking inton this toofor a while and cant seem to figure it out.
0x0F is just letting both fans all out and 0x14 controls the right fan only when 0x0F is set to 08 or any other value that has similar effect. Did anyone find a register responsible for the left fan?

**Comment by NeuroCPP** [comment]
> hey guys ive been looking into this too for a while and cant seem to figure it out.
> 0x0F is just letting both fans all out and 0x14 controls the right fan only when 0x0F is set to 08 or any other value that has similar effect. Did anyone find a register responsible for the left fan?

Regarding the left fan 
I might be having the same issue
I will try to find a possible solution 👍

**Comment by Sayyid01** [comment]
> to be honest, this is also very complicated for me. I tried doing what @dzid26 said, I changed the FF values to 0x0 but it either does nothing or it doesn't accept it and i couldn't do anything else but there is a working config I don't know if you saw it below that you can use to control the fan you can choose the speed, not just 0 or 100 but it makes the GPU fan go to 100% https://www.mediafire.com/file/22s30kjgt5mz5m7/HP_Pavilion_Gaming_Laptop_15-dk1xxx.xml/file you also need to download this dll file [#27 (comment)](https://github.com/UraniumDonut/nbfc-revive/issues/27#issuecomment-967684864) and put it in the nbfc folder in the plugins folder for the program to work properly.

I have an HP Pavillion Gaming 15-dk1064tx with RTX 2060 Max Q, based on the reading, my fan only runs 30% at all times (I don't know why this happens, but it is super quiet tho), though this makes my laptop almost boiling at 90°C when in zoom call and sharing screen. using the config above, it seems both fans have the same speed for me.
